### PR TITLE
Better handle egress destination telemetry

### PR DESCRIPTION
**Pull request by jshaughn**
Egress destination telemetry is lacking and may not be easily fixed on the isti side.  This adds some special case code to help things out for the Kiali graph.

Note that the bad telem is still in Prometheus, so direct metric query won't match up completely unless we want to patch that as well.  That could be a follow up, if needed. 

Closes https://github.com/kiali/kiali/issues/2999

### CODE REVIEW

File: graph/telemetry/istio/util/util.go [hunk 1]
```diff
@@ -10,8 +11,30 @@ import (
...
+	if egressHost == "" {
+		egressHost = fmt.Sprintf("istio-egressgateway.%s.svc.cluster.local", config.Get().IstioNamespace)
+	}
+
+	if destSvc == egressHost && destSvc == destSvcName {
+		istioNs := config.Get().IstioNamespace
+		return istioNs, "istio-egressgateway", istioNs, "istio-egressgateway", "istio-egressgateway", "latest", true
+	}
+
+	return destSvcNs, destSvcName, destWlNs, destWl, destApp, destVer, true
```
[the remark this answers is not in the document]
Comment (jshaughn) [reply graph/telemetry/istio/util/util.go:34]: It should be false, I will fix it. Good catch.

**Comment by israel-hdez** [comment]
So, after enabling "istio-namespace" in the namespaces to graph, I see the expected graph:

image

**Comment by jshaughn** [comment]
@mattmahoneyrh This one may be a little tricky to test as it requires some setup like done by @israel-hdez above.  I think it's normal fo you to test on OS, which is good, would like to see if there is a difference with what Edgar saw with minikube.  Ping me if you need some help in testing this.  Thanks...

**Comment by jshaughn** [comment]
> Suggest to add the namespace to the node label so that it can be easily seen to which namespace the node resides.

Yes, I will generate a separate, UI-side PR.  This can be merged as we now understand why we have two service entry nodes, because they are in different namespaces.

**Comment by jshaughn** [comment]
Adding 1.18 backport label as there is an istio customer looking for this fix yet constrained to the packaged version with istio 1.6.

**Pull request merged**